### ISSUE TITLE
setdiff1d (from tensorflow.python.ops.array_ops) is deprecated and will be removed after 2018-11-30

### ISSUE THREAD

**Issue by beginner401** [keras-team/keras#15717]
**System information**.
Ubuntu 20.04.2 LTS (GNU/Linux 5.4.0-74-generic x86_64)
TensorFlow 2.0.0 installed by anaconda
gcc (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0
NVIDIA-SMI 460.80 Driver Version: 460.80 CUDA Version: 11.2

**Describe the problem**.
When I'm using reduce_prod while training, sometimes the warning occurs like that 'setdiff1d (from tensorflow.python.ops.array_ops) is deprecated and will be removed after 2018-11-30'. I review my code and found if I input a tensor with shape like [None, 1, 1, C] into reduce_prod(dim=3, keep_dims=True), the bug will occur. It must have some relation with reduce_prod, for if I remove this layer, the warning will not appear.

**Describe the current behavior**.
When I'm using reduce_prod while training, sometimes the warning occurs like that 'setdiff1d (from tensorflow.python.ops.array_ops) is deprecated and will be removed after 2018-11-30'.

**Describe the expected behavior**.
Maybe the code behind should no more use setdiff1d.

**Standalone code to reproduce the issue**.
Here is a network structure in json that will produce the bug:
{
    "input_shape": [28, 28, 1], 
    "network": [
        {"name": "SeparableConv2d", "params": {"in_channels": 1, "depth_multiplier": 157, "out_channels": 135, "kernel_size": [1, 1], "stride": 3, "dilation": 1, "padding": "valid", "bias": false}}, 
        {"name": "ReduceProd", "params": {"keep_dims": true, "dim": 3, "tensor_space": 4}}, 
        {"name": "Flatten"}, 
        {"name": "Dense", "params": {"in_features": 1, "out_features": 10}}, 
        {"name": "Softmax"}
    ]
}
The ReduceProd is encapsulated using tf.keras.layers.Layer, and it seems that only inference process cannot trigger that warning, but if  I try to train that model, the warning appears. 

The training process is implemented using standardized code for 'with tf.GradientTape() as tape:'.

**Source code / logs**.
setdiff1d (from tensorflow.python.ops.array_ops) is deprecated and will be removed after 2018-11-30.
Instructions for updating:
This op will be removed after the deprecation date. Please switch to tf.sets.difference().

**Comment by chunduriv** [comment]
@beginner401, Can you take a look at[ this issue](https://github.com/tensorflow/tensorflow/issues/42909#issuecomment-686710157) which discusses about the similar issue and let us know if it helps? Thanks!

**Comment by beginner401** [comment]
@chunduriv Oh, thanks for your support and I'm sure this problem is triggered by the same bug, so I will close this issue.
